### PR TITLE
feat(FN-3567): screenreader alternative for grouped column rows

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
The grouped rows of the premium payments table are not very accessible to screenreader users, with lots of blank cells and it does not convey the visual grouping. This PR addresses that by implementing the screenreader navigation order detailed on the ticket.

## Resolution :heavy_check_mark:
- Hide sub rows of grouped rows from assistive technology
- Hide first four visual rows of table from assistive technology
- Display a visually hidden column which contains first four column data in a list for assistive technology

### CODE REVIEW

File: trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts [hunk 1]
```diff
@@ -73,65 +73,89 @@ describe(component, () => {
...
+    ${'Reported Fees'}     | ${'[data-cy="premium-payments-table--reported-fees-header"]'}
+    ${'Reported payments'} | ${'[data-cy="premium-payments-table--reported-payments-header"]'}
+  `("should set the '$columnName' column header to be aria-hidden", ({ selector }: { selector: string }) => {
+    const wrapper = getWrapper();
+
+    wrapper.expectElement(selector).toHaveAttribute('aria-hidden', 'true');
   });
 
-  it("should use the 'govuk-table__header--numeric' class for numeric columns", () => {
+  it('should render the alternative hidden header as visually hidden', () => {
```
Comment (ttbarnes) [inline trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts:100]:
```suggestion
  it('should render the visually hidden alternative header', () => {
```

File: trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts [hunk 2]
```diff
@@ -488,16 +583,15 @@ describe(component, () => {
...
+    const firstRowCellSelector = `${firstRowSelector} [data-cy="total-payments-received"]`;
+    wrapper.expectText(firstRowCellSelector).toRead(totalPaymentsReceived.formattedCurrencyAndAmount!);
+    wrapper.expectElement(firstRowCellSelector).hasClass(numericCellClass);
+    wrapper.expectElement(firstRowCellSelector).toHaveAttribute('data-sort-value', totalPaymentsReceived.dataSortValue.toString());
 
     otherIds.forEach((id) => {
       const rowSelector = `[data-cy*="premium-payments-table-row--feeRecordId-${id}"]`;
       wrapper.expectElement(`${rowSelector}`).toExist();
-      wrapper.expectElement(`${rowSelector} td:contains("${totalPaymentsReceived.formattedCurrencyAndAmount}")`).notToExist();
+      wrapper.expectText(`${rowSelector} [data-cy="total-payments-received"]`).toRead('');
```
Comment (ttbarnes) [inline trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts:594]: Could we split this test up into different files? Lots going on and now has 751 lines

File: trade-finance-manager-ui/templates/utilisation-reports/_macros/premium-payments-table.njk [hunk 3]
```diff
@@ -43,14 +46,15 @@
...
+        {{ tableHeader({ headerText: 'Reported fees', isNumericColumn: true, enableSorting: false, ariaHidden: "true", dataCy: 'premium-payments-table--reported-fees-header' }) }}
+        {{ tableHeader({ headerText: 'Reported payments', isNumericColumn: true, enableSorting: false, ariaHidden: "true", dataCy: 'premium-payments-table--reported-payments-header' }) }}
+        {{ tableHeader({ headerText: 'Total reported payments', isNumericColumn: true, enableSorting: true, ariaSort: 'ascending', dataCy: 'premium-payments-table--total-reported-payments-header' }) }}
+        {{ tableHeader({ headerText: 'Payments received', isNumericColumn: true, enableSorting: false, dataCy: 'premium-payments-table--payments-received-header' }) }}
+        {{ tableHeader({ headerText: 'Total payments received', isNumericColumn: true, enableSorting: enablePaymentsReceivedSorting, ariaSort: 'none', dataCy: 'premium-payments-table--total-payments-received-header' }) }}
+        {{ tableHeader({ headerText: 'Status', isNumericColumn: false, enableSorting: true, ariaSort: 'none', dataCy: 'premium-payments-table--status-header' }) }}
```
Comment (ttbarnes) [inline trade-finance-manager-ui/templates/utilisation-reports/_macros/premium-payments-table.njk:57]: Could we add an empty line in between each `tableHeader` please? Finding it hard to follow

Would be great to indent each param as well i.e

```
{{ tableHeader({
  headerText: 'Facility ID',
  isNumericColumn: false,
  enableSorting: false,
  ariaHidden: "true",
  dataCy: 'premium-payments-table--facility-id-header'
}) }}
```

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts (hunk 1 above)
Comment (RLavender98) [reply trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts:100]: The test above checks it is rendered just not that it is visually hidden, which is what this test is for, so I don't think that suggested name is accurate. But I will make the name more explicit.

**Review by ttbarnes — APPROVED** [review]
LGTM

File: trade-finance-manager-ui/templates/utilisation-reports/_macros/premium-payments-table.njk (hunk 3 above)
Comment (ttbarnes) [reply trade-finance-manager-ui/templates/utilisation-reports/_macros/premium-payments-table.njk]: Thanks @RLavender98 much easier to read etc now 😍

**Review by sw-mattw — APPROVED** [review]
Nice addition
